lib/m2ts/caption-stream.js

```diff
@@ -17,128 +17,8 @@
-  * @see Rec. ITU-T H.264, 7.3.2.3.1
-  */
-var parseSei = function(bytes) {
-  var
-    i = 0,
-    result = {
-      payloadType: -1,
-      payloadSize: 0
-    },
-    payloadType = 0,
-    payloadSize = 0;
-
-  // go through the sei_rbsp parsing each each individual sei_message
-  while (i < bytes.byteLength) {
-    // stop once we have hit the end of the sei_rbsp
-    if (bytes[i] === RBSP_TRAILING_BITS) {
-      break;
-    }
-
-    // Parse payload type
-    while (bytes[i] === 0xFF) {
-      payloadType += 255;
-      i++;
-    }
-    payloadType += bytes[i++];
-
-    // Parse payload size
-    while (bytes[i] === 0xFF) {
-      payloadSize += 255;
-      i++;
-    }
-    payloadSize += bytes[i++];
-
-    // this sei_message is a 608/708 caption so save it and break
-    // there can only ever be one caption message in a frame's sei
-    if (!result.payload && payloadType === USER_DATA_REGISTERED_ITU_T_T35) {
-      result.payloadType = payloadType;
-      result.payloadSize = payloadSize;
-      result.payload = bytes.subarray(i, i + payloadSize);
-      break;
-    }
-
-    // skip the payload and parse the next message
-    i += payloadSize;
-    payloadType = 0;
-    payloadSize = 0;
-  }
-
-  return result;
-};
-
-// see ANSI/SCTE 128-1 (2013), section 8.1
-var parseUserData = function(sei) {
-  // itu_t_t35_contry_code must be 181 (United States) for
-  // captions
-  if (sei.payload[0] !== 181) {
-    return null;
-  }
-
-  // itu_t_t35_provider_code should be 49 (ATSC) for captions
-  if (((sei.payload[1] << 8) | sei.payload[2]) !== 49) {
-    return null;
-  }
-
-  // the user_identifier should be "GA94" to indicate ATSC1 data
-  if (String.fromCharCode(sei.payload[3],
-                          sei.payload[4],
-                          sei.payload[5],
-                          sei.payload[6]) !== 'GA94') {
-    return null;
-  }
-
-  // finally, user_data_type_code should be 0x03 for caption data
-  if (sei.payload[7] !== 0x03) {
-    return null;
-  }
-
-  // return the user_data_type_structure and strip the trailing
-  // marker bits
-  return sei.payload.subarray(8, sei.payload.length - 1);
-};
-
-// see CEA-708-D, section 4.4
-var parseCaptionPackets = function(pts, userData) {
-  var results = [], i, count, offset, data;
-
-  // if this is just filler, return immediately
-  if (!(userData[0] & 0x40)) {
-    return results;
-  }
-
-  // parse out the cc_data_1 and cc_data_2 fields
-  count = userData[0] & 0x1f;
-  for (i = 0; i < count; i++) {
-    offset = i * 3;
-    data = {
-      type: userData[offset + 2] & 0x03,
-      pts: pts
-    };
-
-    // capture cc data when cc_valid is 1
-    if (userData[offset + 2] & 0x04) {
-      data.ccData = (userData[offset + 3] << 8) | userData[offset + 4];
-      results.push(data);
-    }
-  }
-  return results;
-};
+var Stream = require('../utils/stream');
+var cea708Parser = require('../tools/cea708-parser');
 
 var CaptionStream = function() {
```

Comment:
I agree with you on this, and we should have had the conversation in a more open room, but the general consensus was to leave them as they are since it may be more confusing when LHLS is merged that things moved around. The eventual plan is to move them to `tools` though!